### PR TITLE
New reference for using Redpanda tuning capabilities

### PR DESCRIPTION
**Pull request by greg-szrama**
Resolves [Issue #2012](https://app.zenhub.com/workspaces/documentation-628ba1cf6f2f640015ffe494/issues/gh/redpanda-data/documentation-private/2012)

This guide discusses the autotuner and lists the categories of software configurations managed by Redpanda.

[Preview](https://deploy-preview-305--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/self-hosted/manual/linux-system-tuning/).

### CODE REVIEW

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
Redpanda includes numerous features to tune your Linux systems for optimal performance. You can trigger hardware-based optimizations through the `rpk redpanda tune` command, also called the autotuner. Software-based configurations are managed automatically through the `systemd` suite of tools.
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
You can invoke the Redpanda autotuner using the `rpk redpanda tune` command. This command may optimize both native Linux and Kubernetes nodes. It analyzes the hardware configuration of the worker node and sets appropriate kernel options. You can also specify some hardware parameters using `ulimit` or other system tools while using the autotuner to manage others.
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
The Redpanda autotuner has a large array of hardware tuners to choose from. You may choose to either execute all available tuners, or to apply only a subset of them. This is particularly important when you wish to maintain specific configurations and prevent Redpanda from altering these settings.
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
For the full list of available tuners, refer to the xref:reference:rpk/rpk-redpanda/rpk-redpanda-tune.adoc[`rpk redpanda tune` guide]. The autotuner is also available for managing Kubernetes nodes, but note that changes to the Linux kernel for running Kubernetes images are not persisted. For more information on Kubernetes configuration and execution, refer to xref:deploy:deployment-option/self-hosted/kubernetes/k-tune-workers.adoc[Kubernetes worker node tuning].
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
While Redpanda provides numerous metrics, the configurations applied using the autotuner and systemd are not exposed internally. You should continue monitoring your system health through standard means. For example, the systemd configuration sets the file handle limit to 800,000 references. If you are concerned about approaching this limit, you should monitor it through standard system monitoring, such as the use of the `lsof` utility.
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
* Using systemd also offers some support to detect hung services and infinite looping scenarios. This helps ensure you don't have blocking services or processes that may interfere with Redpanda's stream processing.
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
* Redpanda configures asynchronus input/output interfaces and scheduler affinity. This is particularly critical when you rely on real-time processing of data streams for your applications.
```

**Comment by Feediver1** [comment]
Will circle back after SME reviews/approval to see if any additional edits need to be reviewed.

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (ballard26) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: We offer a non-systemd version of crash-loop detection internally via https://github.com/redpanda-data/redpanda/issues/6569 as well. Not sure if its worth mentioning here though.

Comment (ballard26) [reply modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: Upon re-reading this seems to be talking about non-Redpanda processes running on a node with Redpanda. The crash-loop detection I mentioned above only applies to the Redpanda process its self.

**Review by ballard26 — APPROVED** [review]
LGTM. Seems like a good summary of tuning RP. The articles on the auto-tunner its self provide the lions share of details around what exactly RP is looking to tune.

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (mattschumpert) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: @greg-szrama ,

@chrisseto should review/update the last statement here. This and the linked page imply changes get reset on restart (which makes them arguably unusable in K8s clusters), but also @chrisseto leads me to believe its a case of 'some tuners work, others don't'.

Let's be crisp on the real story about k8s with a one line summary here and a couple of sentences in https://docs.redpanda.com/current/deploy/deployment-option/self-hosted/kubernetes/k-tune-workers/#run-the-autotuner that are prescriptive about what tunes should an should not be used and how that works (persistent/not, etc.)

Right now some folks have the impression they just don't work when running on k8s and its more nuanced than that.  let's be as prescriptive as we can

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (mattschumpert) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: @chrisseto  should vet this statement (see comment below)

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (mattschumpert) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: Is this statement really required to teach people how to use the tuners?

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (mattschumpert) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc:28]: @deniscoady what do you think about this statement wrt overall monitoring approach.  Does this really make sense to recommend?

Also, are these configs in the diag bundle?

Wondering if its worth having this paragraph

**Review by mattschumpert — APPROVED** [review]

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (chrisseto) [reply modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: We've been pretty actively trying to pin down some better answer to this. There's a tracking ticket here: https://github.com/redpanda-data/helm-charts/issues/1017#issuecomment-1991847026 and a lot of internal slack discussions.

The current state of affairs is that it's better to assume that tuners within Kubernetes will not work as expected and they should instead be applied at the host node level. As Matt mentions, there's a lot of nuance but I don't think we (or at least I) have a firm enough grasp to suggest anything besides "Run it on the host if you need to rely on it".

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (chrisseto) [reply modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: Yeah I would recommend removing the `and Kubernetes nodes` part. The tuners rely on too many non-Kubernetes friendly (Tuning mounted devices) or Kubernetes inaccessible (Setting up systemd services) things to claim that Kubernetes is well supported or even supported.

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (greg-szrama) [reply modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: Revised the wording here. I want to maintain the link to the K8s tuner so folks know where to go if that's what they're trying to do.

**Review by chrisseto — APPROVED** [review]

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
Redpanda includes several features to tune your Linux systems for optimal performance. You can trigger hardware-based optimizations through the `rpk redpanda tune` command, also called the autotuner. Software-based configurations are managed automatically through the `systemd` suite of tools.
```

Comment (Feediver1) [reply modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: I only saw a few features mentioned--numerous suggests many more...hence, the wording update request here.

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
You can invoke the Redpanda autotuner using the `rpk redpanda tune` command to optimize native Linux nodes. It analyzes the hardware configuration of the worker node and sets appropriate kernel options. You can choose to set certain hardware parameters using `ulimit` or other system tools while using the autotuner to manage others.
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
For the full list of available tuners, refer to xref:reference:rpk/rpk-redpanda/rpk-redpanda-tune.adoc[`rpk redpanda tune`]. When using Kubernetes, you should tune your system at the host node level. For more information on Kubernetes configuration and execution, refer to xref:deploy:deployment-option/self-hosted/kubernetes/k-tune-workers.adoc[Kubernetes worker node tuning].
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
* Using systemd to help detect hung services and infinite looping scenarios. This helps ensure you don't have blocking services or processes that may interfere with Redpanda's stream processing.
```

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc:20]: The wording of the bullets does not really match the construct provided in the introductory statement. I've updated two. Can you please reword this one so that it works similarly? 

- Redpanda configures asynchronus input/output interfaces and scheduler affinity. This is particularly critical when you rely on real-time processing of data streams for your applications.

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]:
```suggestion
* Using systemd slices (see https://www.scylladb.com/2019/09/25/isolating-workloads-with-systemd-slices/[this blog post] for more details), which allows Redpanda to efficiently use system resources while isolating tasks for scalability and performance reasons.
```

**Review by Feediver1 — APPROVED** [review]
Please implement the requested updates. Thx!

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [reply modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc:28]: @greg-szrama Did you get the feedback/guidance from @deniscoady ?

File: modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc
Comment (Feediver1) [reply modules/deploy/pages/deployment-option/self-hosted/manual/linux-system-tuning.adoc]: @greg-szrama I noticed this content is still in, but did not see an answer to Matt's question. Was there one?

**Pull request merged**
